Lib/multiprocessing/pool.py

```diff
@@ -845,3 +872,6 @@ def _help_stuff_finish(inqueue, task_handler, size):
             pass
         for i in range(size):
             inqueue.put(None)
+
+    def _wait_for_updates(self, timeout):
+        time.sleep(timeout)
```

Comment:
I'm now curious: would it be possible to use something smarter to avoid a sleep?

Reply:
We would need a pipe-based queue (or a raw pipe) to notify a select-based solution. The queues used by the `ThreadPool` are `queue.SimpleQueue()` and are not file-descriptor based so we cannot wait on them on `select`.

We could refactor the general queue to allow a custom notifier in every change but I prefer to do that in another PR.

Reply:
Let's keep focused and concentrate on `Pool`. The `ThreadPool` isn't very much used.

Reply:
> Let's keep focused and concentrate on Pool. The ThreadPool isn't very much used.

Sure.